EventFilter/SiStripRawToDigi/plugins/SiStripRawToDigiUnpacker.cc

```diff
@@ -1048,7 +815,7 @@ namespace sistrip {
 
```

Comment:
static analyzer is complaining about const cast away on L812
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-5e6c19/4284/llvm-analysis/report-cd0929.html#EndPath
It looks like ```const TFHeaderDescription* header = (const TFHeaderDescription*)data_u32;``` or an appropriate cast will fix the warning.

I have this and a few other const_cast cleanups in this package in 
slava77:CMSSW_11_1_0_pre1/test28756/pull28756-31d9c24 https://github.com/cms-sw/cmssw/commit/98b95af12c91c77f2bcce2e271acfe8dba452d48

Reply:
Thanks! I cherry-picked the commit, but implementing the const getter with the non-const version resulted in an infinite recursion, so I went back to doing it the other way around https://github.com/cms-sw/cmssw/pull/28756/commits/f0c4e61934314a36230ed2394b440374fcd726de (using `std::as_const`, as suggested in https://stackoverflow.com/a/47369227/11079521)

Reply:
I only checked that it compiles. Thank you for checking the full details.
I suspect that the static analyzer will still complain here. I guess we can (later?) add a note "CMS_SA_ALLOW" in front of the implementation line.
@makortel @Dr15Jones do you have a better suggestion?

Reply:
so, the static analyzer indeed does not like it; but based on https://stackoverflow.com/a/47369227/11079521, it looks like the pattern is coming from recommended literature.
I consider this an acceptable solution then.

Reply:
> but based on https://stackoverflow.com/a/47369227/11079521, it looks like the pattern is coming from recommended literature.
> I consider this an acceptable solution then.

I agree.

The static analyzer warning can indeed be silenced once(/if) we get the static analyzer to support applying `CMS_SA_ALLOW` to `const_cast`.